src/middlewared/middlewared/service/config_service.py

```diff
@@ -29,17 +35,62 @@ def __new__(cls, name, bases, attrs):
 
         namespace = klass._config.namespace.replace('.', '_')
         config_entry_key = f'{namespace}_entry'
-
-        if klass.ENTRY == NotImplementedError:
-            klass.ENTRY = Dict(config_entry_key, additional_attrs=True)
-
-        config_entry_key = klass.ENTRY.name
-
-        config_entry = copy.deepcopy(klass.ENTRY)
-        config_entry.register = True
-        klass.config = returns(config_entry)(klass.config)
+        config_model_name = f'{namespace.capitalize()}Config'
+
+        if klass._config.entry is not None or klass.ENTRY == NotImplementedError:
+            # FIXME: We do this temporarily to avoid breaking existing code
+            klass.ENTRY = None
+            if klass._config.entry is None:
+                klass._config.entry = create_model(
+                    f'{namespace.capitalize()}Entry',
+                    __base__=(AllowExtraBaseModel,),
```

Comment:
Why should extra fields be allowed?

Reply:
If entry is not defined, in that case by default we will allow any fields to be there. This is not what we want in the final stage of our api but when there is no entry defined i.e no schema in place - then that is what we would be going with to make sure the new service works.